app/scripts/presenters/visit_presenter.js

```diff
@@ -22,7 +30,15 @@ define([
     },
 
     imageUrl: function () {
-      return this.hasValidExtractedImageUrl() ? this.extractedImageUrl : this.screenshot_url;
+      var url = this.hasValidExtractedImageUrl() ? this.extractedImageUrl : this.screenshot_url;
+
+      // it would be smarter to only crop screenshots if we're not in retina, but it's easier
+      // to let the imageProxy just handle it for now.
+      return imageProxy.crop(url, IMAGE_DISPLAY_WIDTH, IMAGE_DISPLAY_HEIGHT);
+    },
+
+    imagePosition: function () {
+      return (this.title.length % 2) ? 'left' : 'right';
```

Comment:
I think the _code_ is readable. I'd question the _logic_ of alternating the floats of the images though. Seems like it'll lead to an inconsistent UI which can be a bit of a drag when things appear randomly shuffled. Hurts my muscle memory.